src/Illuminate/Support/Collection.php

```diff
@@ -1076,14 +1076,18 @@ public function forPage($page, $perPage)
     /**
      * Partition the collection into two arrays using the given callback or key.
      *
-     * @param  callable|string  $callback
+     * @param  callable|string  $key
+     * @param  mixed  $operator
+     * @param  mixed  $value
      * @return static
      */
-    public function partition($callback)
+    public function partition($key, $operator = null, $value = null)
     {
         $partitions = [new static, new static];
 
-        $callback = $this->valueRetriever($callback);
+        $callback = func_num_args() == 1
```

Comment:
Shouldn't it be `===`?

Reply:
It will not make any difference in this case.
func_num_args() will always return an integer, no real need to typecheck in this case.

Reply:
Technically there's no difference indeed, just like 90% of other strict comparison cases out there. The difference here is we'll have an (arguably) better code standard.